### DISCUSSION TITLE
Status code 409 per ClientsApi#modifyClient

### DISCUSSION THREAD

**Discussion by ambra886** [fattureincloud/api#267] [Q&A] [answered]
Salve, 

stiamo cercando di tenere aggiornati i dati dei clienti con la versione 2 della Api.
Non ci è chiaro perchè otteniamo lo status 409 con messaggio "Cliente già esistente."  in seguito alla richiesta di modifica del cliente. Stiamo usando la Java SDK versione 2.0.14.
Il flusso è questo:

- Ricerca cliente per codice fiscale o partita iva
- Se troviamo il cliente recuperiamo l'ID 
- Creiamo la richiesta di modifica

Grazie

**Comment by valmoz** [comment, marked as answer]
Buongiorno @ambra886 
il controllo di univocità del cliente è descritto qui: [Client Creation Guide](https://developers.fattureincloud.it/docs/guides/client-creation#%EF%B8%8F-client-uniqueness)
Le torna quanto descritto?

A disposizione
Mauro

**Reply by ambra886** [reply]
Buona sera @valmoz

in inserimento si. Il problema è che il messaggio ci compare con la chiamata di modifica, abbiamo provato a settare l'id anche nel body della richiesta e non solo come path variable ma non è servito. Nel body ci sono sia il vat_number che il tax_code ma non sono cambiati, il code invece è vuoto.

Vi ringraziamo ancora.

**Reply by emajo** [reply]
Ciao @ambra886, quell'errore viene ottenuto in modifica in caso i criteri di univocità del cliente non vengano effettivamente rispettati, per esempio, abbiamo un cliente del tipo:
```
CODE: "",
TAX_CODE: "1234",
VAT_NUMBER: "", 
```

ipotizzando una modifica che va a settare vat_number="4321" in caso esista un altro cliente del tipo
 
```
CODE: "",
TAX_CODE: "1234",
VAT_NUMBER: "4321", 
```

 verrà restituito un 409.
 
 I criteri i univocità sono spiegati [qui](https://developers.fattureincloud.it/docs/guides/client-creation#%EF%B8%8F-client-uniqueness), ci faccia sapere se è tutto chiaro e se riesce a risolvere.
 
 Grazie,
 Emanuele.

**Reply by ambra886** [reply]
Buon pomeriggio,

grazie delle indicazioni. Abbiamo scoperto il problema. Il cliente aveva anagrafiche doppie una con tax_code e vat_number e una solo con tax_code. Eliminando quella con il solo tax_code l'aggiornamento funziona correttamente.
Grazie
